decidim-core/app/controllers/decidim/devise/sessions_controller.rb

```diff
@@ -8,7 +8,9 @@ class SessionsController < ::Devise::SessionsController
       layout "application"
 
       def after_sign_in_path_for(user)
-        if user.is_a?(User) && user.sign_in_count == 1 && Decidim.authorization_handlers.any?
+        if user.role?(:admin)
```

Comment:
What about process admins?

What I though actually, is to check if there's a redirect in the session/cookie/wherever devise sets it and if it's present redirect there. I _think_ that when you access `/admin` without being logged, this is stored somewhere so you get redirected there after login (with Devise default behaviour).